src/lib/statistics/table_statistics.hpp

```diff
@@ -1,75 +1,41 @@
 #pragma once
 
+#include <atomic>
+#include <iostream>
 #include <memory>
+#include <optional>
+#include <unordered_map>
 #include <vector>
 
-#include "all_parameter_variant.hpp"
 #include "all_type_variant.hpp"
-#include "types.hpp"
 
 namespace opossum {
 
 class BaseColumnStatistics;
+class Table;
 
 /**
- * Statistics about a table, with algorithms to perform cardinality estimations.
+ * Container for all cardinality estimation statistics gathered about a Table. Also used to represent the estimation of
+ * a temporary Table during Optimization.
  */
-class TableStatistics final {
+class TableStatistics {
  public:
-  // magic numbers below are taken from paper "Access path selection in a relational database management system",
-  // P. Griffiths Selinger, 1979
-  static constexpr auto DEFAULT_LIKE_SELECTIVITY = 0.1f;
-  static constexpr auto DEFAULT_OPEN_ENDED_SELECTIVITY = 1.f / 3.f;
-  // Made up magic number
-  static constexpr auto DEFAULT_DISJUNCTION_SELECTIVITY = 0.2f;
-
-  TableStatistics(const TableType table_type, const float row_count,
-                  const std::vector<std::shared_ptr<const BaseColumnStatistics>>& column_statistics);
-  TableStatistics(const TableStatistics& table_statistics) = default;
-
   /**
-   * @defgroup Member access
-   * @{
+   * Generates histograms for all columns
```

Comment:
It should become clearer that this is a design decision

Reply:
```c++

  /**
   * Currently, we generate EqualDistinctCountHistograms for all columns and determine their null value ratio.
   * Histograms are used because they are currently our best option to estimate predicate and join cardinalities during
   * optimization.
   */
  static std::shared_ptr<TableStatistics> from_table(const Table& table);

```